packages/thyra/adapters/epetra/src/Thyra_EpetraThyraWrappers.cpp

```diff
@@ -786,25 +812,27 @@ Thyra::get_Epetra_MultiVector(
   )
 {
   using Teuchos::rcpWithEmbeddedObj;
-  using Teuchos::rcpFromRef;
   using Teuchos::ptrFromRef;
   using Teuchos::ptr_dynamic_cast;
   using Teuchos::outArg;
+
+  double* rawMvData = NULL;
+  Ordinal mvLeadingDim = 0;
+
+  Ptr<SpmdMultiVectorBase<double> > mvSpmdMv =
+    ptr_dynamic_cast<SpmdMultiVectorBase<double> >(ptrFromRef(mv));
   ArrayRCP<double> mvData;
-  Ordinal mvLeadingDim = -1;
-  Ptr<SpmdMultiVectorBase<double> > mvSpmdMv;
-  if (nonnull(mvSpmdMv = ptr_dynamic_cast<SpmdMultiVectorBase<double> >(ptrFromRef(mv)))) {
+  if (nonnull(mvSpmdMv)) {
     mvSpmdMv->getNonconstLocalData(outArg(mvData), outArg(mvLeadingDim));
+    rawMvData = mvData.getRawPtr();
   }
-  if (nonnull(mvData)) {
-    return rcpWithEmbeddedObj(
-      new Epetra_MultiVector(
-        ::View, map, mvData.getRawPtr(), mvLeadingDim, mv.domain()->dim()
-        ),
-      mvData
-      );
-  }
-  return ::Thyra::get_Epetra_MultiVector(map, rcpFromRef(mv));
+
+  return rcpWithEmbeddedObj(
+    new Epetra_MultiVector(
+      ::View, map, mvData.getRawPtr(), mvLeadingDim, mv.domain()->dim()
```

Comment:
Okay, got it.  If there are no elements on the process, we still create the Epetra_MultiVector object, we just pass in null (which is what is turned by default from default constructed `mvData`).